libbeat/cmd/instance/beat.go

```diff
@@ -248,7 +248,14 @@ func NewBeat(name, indexPrefix, v string, elasticLicensed bool) (*Beat, error) {
 	if err != nil {
 		return nil, fmt.Errorf("failed to get host information: %w", err)
 	}
-	fqdn := h.Info().FQDN
+
+	fqdn, err := h.FQDN()
+	if err != nil {
+		// FQDN lookup is "best effort".  We log the error, fallback to
+		// the OS-reported hostname, and move on.
+		logp.NewLogger(name).Infof("unable to lookup FQDN: %s, using hostname = %s", err.Error(), hostname)
```

Comment:
I'm not able to get this log message to be printed to the logs.  After reading the godoc for the `logp.NewLogger` function, I've also tried using `logp.L()` instead but that didn't work either.  Adding a `fmt.Println` right above this line works, though.  @faec is helping me figure out why the logging might not be working here.

Reply:
Turns out logging isn't configured at this point so we need to move the FQDN initialization to slightly later in the execution.